### DISCUSSION TITLE
Appending multiple tables with incremental refresh in tabular editor not functioning

### DISCUSSION THREAD

**Discussion by gregorysimpson** [TabularEditor/TabularEditor3#1375] [🙋♀️TE3 Q&A]
All,
I am working on a report where I have 38 different tables coming from 19 different oracle servers.  The report and data model work fine in Power BI desktop and all data appears appended together.  After applying incremental refresh in PBI desktop and publishing I connect via tabular editor to the model.  I begin applying partition policies and updating the partitions using the process below:
1. Apply refresh policy to tables
2. Apply Refresh-Data only to partitions
3. Apply Refresh-Calculate only to individual table  (I can successfully preview all loaded data on the individual tables here)
4. Apply Refresh- (Have tried full, recalc, etc. to appended table that contains all the data)
5. Calculate Model.

I cant no matter what I try get the newly loaded partition data in the appended tables to show in the combined table that drives my report visuals.  What am I missing!!?!?!?!?!?

**Comment by otykier** [comment]
What do you mean by "appended table"? Is it a calculated table that combines the data using a DAX expression like `UNION( table1, table2, table3, ... )`

Any additional details you can share to help us understand how your model is set up would be appreciated. Thanks!

**Comment by gregorysimpson** [comment]
Yes, it is a DAX Union.  Without using the SelectColumns function to order them, the union does not work even though in Power Query all the 38 appended tables have the same columns in the same order.  This leads to a huge, unnecessary DAX expression to create the table.  Without selectcolumns, the data from some columns gets mixed up and goes to the wrong columns.

**Reply by otykier** [reply]
Okay, so I'm trying to understand why this issue would be related to Tabular Editor. Are you seeing an error message when trying to perform the calculate refresh? What is the "State" of the Calculated Table? I.e. are you seeing any output in the **Messages** window, similar to the following:

![image](https://github.com/user-attachments/assets/0936e8a9-83fc-411d-903e-b76a62b835cf)

In case the calculated table *has* been refreshed, what have you done to verify the validity of the DAX expression of the table?

**Comment by gregorysimpson** [comment]
There is no error message.  The data is just incorrect when using the UNION function for building the calculated table, despite the child tables being correct.  I have validated and revalidated the data and the only way the UNION function will work correctly is with the selectcolumns.  The state is Ready.

Ryan Simpson, P.E. | Eastman
Process Analytics Engineer
Office: +1 423.229.5220 | Mobile: +1 270.302.9131
Member ISA, IFPAC, AICHE

From: Daniel Otykier ***@***.***>
Sent: Tuesday, October 1, 2024 6:16 AM
To: TabularEditor/TabularEditor3 ***@***.***>
Cc: Ryan Simpson (US) ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [TabularEditor/TabularEditor3] Appending multiple tables with incremental refresh in tabular editor not functioning (Discussion #1375)

You don't often get email from ***@***.******@***.***>. Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>
CAUTION: External Email


Okay, so I'm trying to understand why this issue would be related to Tabular Editor. Are you seeing an error message when trying to perform the calculate refresh? What is the "State" of the Calculated Table? I.e. are you seeing any output in the Messages window, similar to the following:

image.png (view on web)<https://github.com/user-attachments/assets/0936e8a9-83fc-411d-903e-b76a62b835cf>

In case the calculated table has been refreshed, what have you done to verify the validity of the DAX expression of the table?

-

**Reply by otykier** [reply]
Again, not sure why you believe this to be a Tabular Editor issue. Tabular Editor just lets you edit the model *metadata*, i.e. table names, expressions, etc. Actual *data* is not handled by Tabular Editor. If you see incorrect data, there must be an issue in your DAX expression or the data that feeds into the calculation.

Either way, it’s not easy to provide guidance without knowing the details (the exact DAX expressions used, the model structure, expected output based on some sample input, etc.). But you may be better served asking the question on the Microsoft Fabric community forum: https://community.fabric.microsoft.com/
